### PR TITLE
Improving node.js doc of the main site

### PR DESCRIPTION
**Pull request by Spangenhelm**

### CODE REVIEW

**Comment by Spangenhelm** [comment]
Hi again @Jermolene this change has not been merged yet either, is it because you think it needs modifications ?
Let me know :)

File: How_to_enable_a_wiki_folder_for_client-server_usage.tid [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+## In the `plugins` section, add those lines:
+#> `"tiddlywiki/tiddlyweb",
+"tiddlywiki/filesystem",`
+# Save the wiki then exit the process in the terminal and re-launch it in order for this take effect.<br /><br />
+# To check if the modifications are functionnals:
+#> Once you have re-launched your wiki, check for the presence of the plugins in the control panel, make sure they are not disabled.
+#>And / or
+#> Check for scrolling text in the opened terminal, under: <br />`Serving on 127.0.0.1:8080 (press ctrl-C to exit)`
+
+@@color: orange; ⚠@@ Annoyingly, the plugin library can't be used to install these components under [[Node.js]], one has to edit the `tiddlywiki.info` file.
```
Comment (sukima) [inline How_to_enable_a_wiki_folder_for_client-server_usage.tid]: I wouldn't say Annoyingly 'cause it suggests that there is a bug that wasn't fixed. The issue is that TiddlyWiki plugins are *not* node modules and so they should not have anything to do with node. That is why they are referenced in a TiddlyWiki specific file not a NPM file. This is as designed.

Also, the tw5.com edition has a `.warning` macro for the purpose of the warning symbol so `@@color: orange; ⚠@@` is unnecessary. See [Documentation Macros](http://tiddlywiki.com/#Documentation%20Macros) for more info.

Comment (Spangenhelm) [inline How_to_enable_a_wiki_folder_for_client-server_usage.tid]: Ok so what about this change for the last line:
<<.warning "The plugin library can't be used to install these components under [[Node.js]], one has to edit the `tiddlywiki.info` file.">>

or do you think that i should add "The issue is that TiddlyWiki plugins are not node modules and so they should not have anything to do with node" ?

Thanks for your advices

Comment (sukima) [inline How_to_enable_a_wiki_folder_for_client-server_usage.tid]: Ahh now I understand what you mean! How about this:

> With Node.JS any plugins you add via the plugin library are **not** persisted across page refreshes. To make it permanent you have to edit the `tiddlywiki.info` file.

Of if you prefer your first option would work also (i.e. "The plugin library can't be used to install these components under [[Node.js]], one has to edit the tiddlywiki.info file").

**Comment by Jermolene** [comment]
Sorry for the delay

It seems that this new tiddler pretty much duplicates the intent of "Installing TiddlyWiki on Node.js". It's chattier, and adds material about the plugin library. It's also a bit confusing; if the user has followed the instructions in "Installing TiddlyWiki on Node.js" then their `tiddlywiki.info` file will already contain the references to `tiddlywiki/tiddlyweb` and `tiddlywiki/filesystem`.

**Comment by Spangenhelm** [comment]
Ok @Jermolene you could have been more explicit: what do we do ? Do you want me to close this one ? I don't mind but simply tell me, thanks

**Comment by Jermolene** [comment]
My apologies @Spangenhelm, I'd be pleased if you could make a new pull request for "Installing TiddlyWiki on Node.js" that largely leaves the existing instructions alone, but adds a section (with heading) at the bottom that explains why and how you'd install an edition other than the default "server" edition.

**Comment by Spangenhelm** [comment]
Hi @Jermolene no problem, should i warn the user that the other versions won't auto-save the changes or do i just add a link to this tiddler (`How_to_enable_a_wiki_folder_for_client-server_usage.tid`) ? By the way i think we should rename this tiddler to "How to add server-related components to an existing wiki"

Also i need help for the explanation about installing other editions: what is the purpose other than contributing to the code or to the docs ?

I have added a heading to the first part of the explanation too to make it more clear.
First part: server-related edition
Second part: installing other editions

Thank you

**Comment by Jermolene** [comment]
Hi @Spangenhelm 

> should i warn the user that the other versions won't auto-save the changes or do i just add a link to this tiddler (How_to_enable_a_wiki_folder_for_client-server_usage.tid) ?

Sorry for the confusion. My suggestion above was that *instead* of adding the new tiddler `How_to_enable_a_wiki_folder_for_client-server_usage.tid` that we would update `Installing TiddlyWiki on Node.js` to include material on installing an edition other than the usual client-server edition.

> Also i need help for the explanation about installing other editions: what is the purpose other than contributing to the code or to the docs ?

Are you asking what is the purpose of the other editions? I think the answer depends on the edition; the language editions, for instance, make a convenient starting point for a new wiki in that language.

Perhaps a related question would be to explain why the tiddlyweb and filesystem plugins are needed?

**Comment by Jermolene** [comment]
btw @Spangenhelm my apologies for the endlessness of this pull request! Async discussions are slow; it would be great to be able to discuss face-to-face.

**Comment by Spangenhelm** [comment]
No problem @Jermolene i have copied the content in the "Installing TiddlyWiki on Node.js" tiddler so i will now close this pull request.

Btw there is something that i found disturbing: the warning macro display a __red__ triangle but that is more the color for errors than warnings don't you think that we should modify it to orange?

@sukima above you said:
> With Node.JS __any__ plugins you add via the plugin library are not persisted across page refreshes. To make it permanent you have to edit the tiddlywiki.info file.

I haven't tried it myself but __any__ plugin will not be persistent after refresh or just the server-related ones ?

Thank you guys for your suggestions!

**Comment by Spangenhelm** [comment]
@Jermolene 

> Perhaps a related question would be to explain why the tiddlyweb and filesystem plugins are needed?

Well i'm not sure... for auto-saving the changes only or is there another reason ?

> it would be great to be able to discuss face-to-face.

I agree but i have never tested : how would we proceed ?

**Comment by Jermolene** [comment]
Hi @Spangenhelm 

> Btw there is something that i found disturbing: the warning macro display a red triangle but that is more the color for errors than warnings don't you think that we should modify it to orange?

I'd be inclined to agree, yes.

> Well i'm not sure... for auto-saving the changes only or is there another reason ?

Indeed, they are responsible for syncing changes between the browser and the server, and between the server and the file system respectively.

> I agree but i have never tested : how would we proceed ?

The easiest possibility is to join one of the regular hangouts; we often discuss documentation issues. Alternatively, persuade @silvyn to organise another TiddlyCamp-FR, and we can get together there!

**Comment by silvyn** [comment]
Thanks @Jermolene :D Hope next year if we can !

@Spangenhelm if you want to join us, you can visit http://forum.tiddlywiki.fr ! I'm sure you can contribute to promote TiddlyWiki and spread the word.

**Comment by Spangenhelm** [comment]
Hi @silvyn thank you for the invitation ! About the upgrade procedure that you talked about in your forum you may not know that you can upgrade via the french empty edition here: http://tiddlywiki.com/languages/fr-FR/empty.html
this way you can enjoy the latest translations (which you can __not__ have with the regular upgrader because it only uses the translation already included within our wiki, if any)

**Comment by Spangenhelm** [comment]
Hi @Jermolene as Sukima said above:

> With Node.JS any plugins you add via the plugin library are not persisted across page refreshes. To make it permanent you have to edit the tiddlywiki.info file.

I haven't tried it myself but __any__ plugin will not be persistent after refresh or just the server-related ones ?

Thank you

**Comment by sukima** [comment]
@Spangenhelm, *Just the server-related ones*.

**Comment by Spangenhelm** [comment]
Thank you @sukima now i am able to submit my new pull request

**Pull request closed without merging**